### PR TITLE
Fix json adding "" to strings when storing

### LINKED ISSUE #182: Bug:  Token is saved with " to SessionStorage
## Steps To Reproduce
I am using the session storage to retrieve the tokens when need to call the API. This returns the token correctly but with extra quotation marks.

Expected token to be returned:
`this is my super secret token`

Returned token:
`"this is my super secret token"`

### PR DESCRIPTION
**Pull request by brumik**
Closes #182 .

### CODE REVIEW

File: src/Hooks.tsx [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import { useEffect, useState } from 'react'
 
+function isString(value: any): boolean {
```
Comment (sebastianvitterso) [inline src/Hooks.tsx:3]:
```suggestion
function isString(value: any): value is string {
```

File: src/Hooks.tsx [hunk 2]
```diff
@@ -23,7 +27,9 @@ function useBrowserStorage<T>(key: string, initialValue: T, type: 'session' | 'l
     try {
       const valueToStore = value instanceof Function ? value(storedValue) : value
       setStoredValue(valueToStore)
-      storage.setItem(key, JSON.stringify(valueToStore))
+
+      const storageString = isString(valueToStore) ? (valueToStore as string) : JSON.stringify(valueToStore);
```
Comment (sebastianvitterso) [inline src/Hooks.tsx:31]: With `value is string` above, we won't need `as string` here.
```suggestion
      const storageString = isString(valueToStore) ? valueToStore : JSON.stringify(valueToStore);
```

**Review by sebastianvitterso — COMMENTED** [review]
I had some comments, but I don't think the issue this works off of is a real problem, so I won't approve until I'm convinced.
As far as I can tell, the issue describes a bug in the auth provider.

**Comment by brumik** [comment]
This is happening with the default setup pkce flow on Keycloak 24 image. I can use the token that I get from the network call. The added " is clearly coming from storing in the session with stringify. I understand that I could just JSON.parse the string from the session storage, but that seems like rather a bug than a feature. If it is in fact a feature then at least it should be documented.

**Comment by sebastianvitterso** [comment]
Let's use the issue to discuss the bug :v:

**Comment by brumik** [comment]
See comment on the original issue #182

**Pull request closed without merging**